File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(str1, str2)
    raise ArgumentError if str1.length != str2.length
    counter=0
    str1.split("").each_with_index do |char, index|
      counter += 1 if char != str2[index]
    end
    counter
  end
end

File: go/hamming/hamming.go
package hamming

import "fmt"

const testVersion = 5

func Distance(str1, str2 string) (int, error) {
	if len(str1) != len(str2) {
		return 0, fmt.Errorf("Strands must be the same length\n%s\n%s", str1, str2)
	}

	counter := 0
	for i := 0; i < len(str1); i++ {
		if str1[i] != str2[i] {
			counter++
		}
	}

	return counter, nil
}
